pkg/network/setup/netstat.go

```diff
@@ -158,14 +159,18 @@ func (c *NetStat) UpdateStatus(vmi *v1.VirtualMachineInstance, domain *api.Domai
 			// Remove the interface from domainInterfaceStatusByMac to mark it as handled
 			if interfaceStatus, exists := domainInterfaceStatusByMac[interfaceMAC]; exists {
 				newInterface.InterfaceName = interfaceStatus.InterfaceName
+				newInterface.InfoSource = interfaceStatus.InfoSource
```

Comment:
Since the merge has one of 3 values:
1. domain
2. guest-agent (can be in this specific line, because we know we are iterating the spec)
3. domain and guest-agent (`InfoSourceDomainAndGA`)

then there are two options left 1 and 3, not just `InfoSourceDomainAndGA`